src/coefficients/linear_coefficient.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef LINEAR_COEF
+#define LINEAR_COEF
+
+#include <memory>
+
+#include "mfem.hpp"
+
+class LinearTransformationCoefficient : public mfem::Coefficient {
+ private:
+  std::shared_ptr<mfem::Coefficient> m_coef;
+  double                             m_scale;
+  double                             m_offset;
+
+ public:
+  LinearTransformationCoefficient(const std::shared_ptr<mfem::Coefficient> coef, const double offset,
+                                  const double scale)
+      : mfem::Coefficient(), m_coef(coef), m_scale(scale), m_offset(offset)
+  {
+  }
+  double Eval(mfem::ElementTransformation &T, const mfem::IntegrationPoint &ip)
+  {
+    double val = m_coef->Eval(T, ip);
+    return m_scale * val + m_offset;
+  };
+};
+
+#endif
```

Comment:
Is this functionality already available in StdFunctionCoefficient?

Reply:
Heh, yep. Well, I need to write a new class similar to `TransformedVectorCoefficient`, but for scalars. I'll do that instead.

Reply:
The reason why I didn't implement `TransformedCoefficient` in serac is because MFEM also has [one by the same name](https://mfem.github.io/doxygen/html/classmfem_1_1TransformedCoefficient.html). 

Most of the time it works sans when you need to capture. I feel like TransformedVectorCoefficient and TransformedCoefficient (better implemented) should be upstreamed except I don't know how MFEM feels about smart pointers.

Reply:
They are unfortunately generally against it. I will ask at the meeting today if that has changed though. One simple way out is to call it `TransformedScalarCoefficient`.

Reply:
Done! I think it's best to have a corresponding API for the scalars and vectors. Plus, then we get the capturing. I'll make a push toward upstreaming these to MFEM.